specs/interop/eth-lockbox.md

```diff
@@ -0,0 +1,255 @@
+
+MUST be triggered when `authorizePortal` is called
+
+```solidity
+event PortalAuthorized(address indexed portal);
+```
+
+#### `LockboxAuthorized`
+
+MUST be triggered when `authorizeLockbox` is called
+
+```solidity
+event LockboxAuthorized(address indexed lockbox);
+```
+
+#### `LiquidityMigrated`
+
+MUST be triggered when `migrateLiquidity` is called
+
+```solidity
+event LiquidityMigrated(address indexed lockbox);
+```
+
+#### `LiquidityReceived`
+
+MUST be triggered when `receiveLiquidity` is called
+
+```solidity
+event LiquidityReceived(address indexed lockbox);
+```
+
+## Invariants
+
+### System level invariants
+
+- The ETH held in the `ETHLockbox` MUST never be less than the amount deposited but not yet withdrawn by the `OptimismPortal`s
+
+- The total withdrawable ETH amount present on all the dependency set's chains MUST NEVER be more than the amount held
+  by the `ETHLockbox` of the cluster
+  > With "withdrawable amount", the ETH balance held on `ETHLiquidity` is excluded
+
+### Contract level invariants
+
+- It MUST allow only authorized portals to lock ETH
+
+- It MUST allow only authorized portals to unlock ETH
+
+- It MUST be in paused state if the `SuperchainConfig` is paused
+
+- No Ether MUST flow out of the contract when in a paused state
+
+- It MUST NOT trigger a new deposit when ETH amount is being unlocked from the `ETHLockbox` by the `OptimismPortal`
+
+- It MUST allow only the `ProxyAdmin` owner to call the `authorizePortal`, `authorizeLockbox` and `migrateLiquidity` functions
+
+- It MUST allow only authorized lockboxes to call the `receiveLiquidity` function
+
+- It MUST migrate the whole ETH liquidity from the source `ETHLockbox` to the destination `ETHLockbox` when calling `migrateLiquidity`
+
+- It MUST emit:
+
+  - An `ETHLocked` event when locking ETH
+
+  - An `ETHUnlocked` event when unlocking ETH
+
+  - A `PortalAuthorized` event when authorizing a portal
+
+  - A `LockboxAuthorized` event when authorizing a lockbox
+
+  - A `LiquidityMigrated` event when migrating liquidity
+
+  - A `LiquidityReceived` event when receiving liquidity
+
+## Architecture
+
+### ETH Management
+
+- ETH is locked in the `ETHLockbox` when:
+
+  - A portal migrates its ETH liquidity when updating
+
+  - A deposit is made with ETH value on an authorized portal
+
+- ETH is unlocked from the `ETHLockbox` when:
+
+  - An authorized portal finalizes a withdrawal that requires ETH
+
+### Merge process
+
+The merge process is the process of migrating the ETH liquidity from one `ETHLockbox` to another.
+
+- The destination `ETHLockbox` MUST call `authorizeLockbox` with the source `ETHLockbox` as argument.
+
+- The source `ETHLockbox` MUST call `migrateLiquidity` with the destination `ETHLockbox` as argument.
+
+- `migrateLiquidity` MUST call `receiveLiquidity` from the destination `ETHLockbox`.
+
+This process ensures that the ETH liquidity is migrated from the source `ETHLockbox` to the correct destination `ETHLockbox`.
```

Comment:
Let's specify if this is intended to be an atomic operation, or separate transactions. Since all involved contracts have the same `adminOwner()` I assume atomic?